grails-app/taglib/org/pih/warehouse/SelectTagLib.groovy

```diff
@@ -426,6 +422,23 @@ class SelectTagLib {
         out << g.select(attrs)
     }
 
+    def selectZoneLocationByLocation = { attrs, body ->
+        def location = Location.get(attrs.id)
+
+        if (location) {
+            attrs.from = Location.findAllByParentLocationAndActive(location, true).sort {
```

Comment:
The same logic should be used for line 322 in LocationController, so let's move this to the service layer and implement it using a criteria query or HQL. You can add the sort / filter in there as well.